core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -245,13 +286,18 @@ public List<TableIdentifier> listTables(Namespace namespace) {
         row ->
             JdbcUtil.stringToTableIdentifier(
                 row.getString(JdbcUtil.TABLE_NAMESPACE), row.getString(JdbcUtil.TABLE_NAME)),
-        JdbcUtil.LIST_TABLES_SQL,
+        JdbcUtil.LIST_TABLE_SQL,
         catalogName,
-        JdbcUtil.namespaceToString(namespace));
+        JdbcUtil.namespaceToString(namespace),
+        "TABLE");
   }
 
   @Override
   public void renameTable(TableIdentifier from, TableIdentifier to) {
+    if (viewExists(to)) {
```

Comment:
Similar to the view rename, we should either be preemptively checking for both or neither to be consistent.  This isn't an atomic check anyway, so the real check is being handled by the database constraints.